src/components/suggestion/areas/song-information.area.tsx

```diff
@@ -30,6 +60,43 @@ const SongInformationArea = ({ proceedToNextArea, onFormSuccess }: SongInformati
     if (!isSongInformationInvalid(watch)) proceedToNextArea()
   }
 
+  const debouncedHandleSearch = debounce((value: string) => {
+    if (value.length == 0) return
+
+    setFetchingSongs(true)
+    getSpotifySearchResults(basePath, getValues().title)
+      .then(async (response) => {
+        const data: SpotifySearchItem[] = ((await response.json()).tracks.items)
+        const items: SearchItem[] = data.map((item) => ({
+          id: item.id,
+          title: item.name,
+          artists: item.artists.map((artist): string => artist.name),
+          link: item.external_urls.spotify
+        }))
+        setSearchResults(items)
+      })
+      .catch(() => setShowSearchError(true))
+      .finally(() => setFetchingSongs(false))
+  }, 400)
+
+  const handleSearch = (value: string) => {
+    setValue("title", value)
+    debouncedHandleSearch(value)
```

Comment:
Did `debounce(handleSearch(value), 400)` not work?

Reply:
no bc i need the `event` from the onChange and `(event) => debounce(..` doesnt work it, you have to call debounce right away like `onChange={debounce(... ` or `const example = debounce(..`